### PR TITLE
Add documentation for standalone SAML authentication in engine

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/2972

### CODE REVIEW

File: source/site-administrators/engine/engine-site-security-guide.rst [hunk 1]
```diff
@@ -12,6 +12,7 @@ The following guide will help you configure Crafter Engine and Crafter Profile t
 #.  Enable security for your website.
 #.  Add authentication to your site, including Facebook login and Single Sign-On.
 #.  Add authorization so that access to certain pages and URLs of your site are restricted.
+#.  Add standalone authentication to your site
```
Comment (sumerjabri) [inline source/site-administrators/engine/engine-site-security-guide.rst]: Remove this line

File: source/site-administrators/engine/engine-site-security-guide.rst [hunk 2]
```diff
@@ -12,6 +12,7 @@ The following guide will help you configure Crafter Engine and Crafter Profile t
 #.  Enable security for your website.
 #.  Add authentication to your site, including Facebook login and Single Sign-On.
```
Comment (sumerjabri) [inline source/site-administrators/engine/engine-site-security-guide.rst]: Create a top-level  "Add SAML 2 Single Sign-on" bullet.

File: source/site-administrators/engine/engine-site-security-guide.rst [hunk 3]
```diff
@@ -442,3 +443,45 @@ expression evaluates to false, access is denied. The following expressions can b
 *   ``hasAnyRole({'role1', 'role2'})``
 *   ``permitAll()``
 *   ``denyAll()``
+
+-------------------------
```
Comment (sumerjabri) [inline source/site-administrators/engine/engine-site-security-guide.rst]: Move this into the single sign-on section. Put this as the first second and call it "SAML 2 Single Sign-On without Crafter Profile" and put it above the SSO with Profile section.

File: source/site-administrators/engine/engine-site-security-guide.rst [hunk 4]
```diff
@@ -442,3 +443,45 @@ expression evaluates to false, access is denied. The following expressions can b
...
+Standalone Authentication
+-------------------------
+
+You can also integrate SAML 2.0 authentication without using Crafter Profile, in this case the requests will still use
```
Comment (sumerjabri) [inline source/site-administrators/engine/engine-site-security-guide.rst]: Don't reference Crafter Profile in this section.

File: source/site-administrators/engine/engine-site-security-guide.rst [hunk 5]
```diff
@@ -7,438 +7,523 @@
...
+        metadata file retrieved from the IdP.
+
+.. note::
+  It’s important to remember that the environment variables set by auth_mellon and used to create this headers depend
+  in the IdP, so you’ll need to check first what the IdP is sending before defining the headers.
+
+Configure your site
+===================
+
+To enable SAML 2.0 in your site configuration (in Studio, Config > site.xml) add or update the security elements as
```
Comment (alhambrav) [inline source/site-administrators/engine/engine-site-security-guide.rst]: Should `Config > site.xml` be `Config > site-config.xml`?

File: source/site-administrators/engine/engine-site-security-guide.rst [hunk 6]
```diff
@@ -7,438 +7,523 @@
...
+Restrict URLs
+=============
 
-                emailFactory = new EmailFactoryImpl()
-                emailFactory.mailSender = mailSender
-                emailFactory.freeMarkerConfig = freeMarkerConfig
-            }
+Sometimes it is not enough to restrict a single page. Sometimes you need to restrict an entire site subtree, or 
+restrict several static assets. For this, Crafter CMS provides configuration parameters that allow you to restrict 
+access based on URL patterns. You just need to add configuration similar to the following in Config > site.xml:
```
Comment (alhambrav) [inline source/site-administrators/engine/engine-site-security-guide.rst]: Should `Config > site.xml` be `Config > site-config.xml`?

File: source/site-administrators/engine/engine-site-security-guide.rst [hunk 7]
```diff
@@ -7,438 +7,523 @@
...
+    Profile Admin Console, select the Tenant and then add the attribute.
 
-Restrict URLs
-=============
+    .. image:: /_static/images/connections_attribute.png
 
-Sometimes it is not enough to restrict a single page. Sometimes you need to restrict an entire site subtree, or restrict several static
-assets. For this, Crafter CMS provides configuration parameters that allow you to restrict access based on URL patterns. You just need
-to add configuration similar to the following in Config > site.xml:
+#.  Add the Facebook appSecret and appKey to your site's config (in Studio, Config > site.xml), like this:
```
Comment (alhambrav) [inline source/site-administrators/engine/engine-site-security-guide.rst]: Should `Config > site.xml` be `Config > site-config.xml`?

File: source/site-administrators/engine/engine-site-security-guide.rst [hunk 8]
```diff
@@ -7,438 +7,523 @@
...
+#.  Now that you have created the access token, you need to "tell" Engine to use it in your site. In Admin Console, 
+    click on List Access Tokens in the navigation menu and copy the ID of the token you just created. Then, depending 
+    on the mode Engine is running, add one of the following configurations (preview is ignored because normally 
+    predefined Personas are used, so there's no need to access the Crafter Profile app).
 
-            return "/templates/web/account-verified.ftl"
-        } else {
-            throw new HttpStatusCodeException(400, "Bad request: token param is missing")
-        }
+    *   **Multi-tenant:** You need to add the access token ID to the Config > site.xml in Studio, and deploy the file
```
Comment (alhambrav) [inline source/site-administrators/engine/engine-site-security-guide.rst]: Should `Config > site.xml` be `Config > site-config.xml`?

**Comment by jdrossl** [comment]
@alhambrav about the config, I replaced the filename with the label used in the UI to avoid confusion because `site.xml` is used in Engine and `site-config.xml` is used in Studio

**Pull request merged**